packages/@aws-cdk/aws-fsx/README.md

```diff
@@ -17,8 +17,188 @@
+Linux operating system. It also provides read-after-write consistency and supports file locking.
+
+### Installation
+
+Import to your project:
 
 ```ts
 import fsx = require('@aws-cdk/aws-fsx');
 ```
+
+### Basic Usage
+
+Setup required properties and create:
+
+```ts
+const stack = new Stack(app, 'Stack');
+const vpc = new Vpc(stack, 'VPC');
+const storageCapacity = 1200;
+const lustreConfiguration = {
+  deploymentType: FsxLustreDeploymentType.SCRATCH_2
+};
+
+const fileSystem = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration,
+  storageCapacity,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+```
+
+### Connecting
+
+To control who can access the file system, use the `.connections` attribute. FSx has a fixed default port, so you don't
+need to specify the port. This example allows an EC2 instance to connect to a file system:
+
+```ts
+fileSystem.connections.allowDefaultPortFrom(instance);
+```
+
+### Mounting
+
+To mount an FSx for Lustre file system programmatically, the mount name of the file system needs to be determined. The
+easiest way to do that is with the AWS CLI, which can be written into a User Data script to be run on start-up of your
+instance. The following example can be used for reference on how to bring up a file system and EC2 instance with the
+file system mounted on the instance at start-up:
+
+```ts
+const app = new App();
+const stack = new Stack(app, 'AwsCdkFsxLustre');
+const vpc = new Vpc(stack, 'VPC');
+
+// Create a security group to share across the file system and ec2 instance and open the FSx ports.
+const securityGroup = new SecurityGroup(stack, 'FsxLustreSecurityGroup', {
+  vpc
+});
+securityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(988), 'Default FSx port ingress');
+securityGroup.addEgressRule(Peer.anyIpv4(), Port.tcp(988), 'Default FSx port egress');
+
+const lustreConfiguration = {
+  deploymentType: FsxLustreDeploymentType.SCRATCH_2,
+};
+const fs = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration,
+  securityGroup,
+  storageCapacity: 1200,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+
+const inst = new Instance(stack, 'Ec2Instance', {
+  instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
+  machineImage: new AmazonLinuxImage({
+    generation: AmazonLinuxGeneration.AMAZON_LINUX_2
+  }),
+  securityGroup,
+  vpc,
+  vpcSubnets: {
+    subnetType: SubnetType.PUBLIC,
+  }
+});
+
+// Need to give the instance access to read information about FSx to determine the file system's mount name.
+inst.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName("AmazonFSxReadOnlyAccess"));
+
+// Add a dependency on the file system to the instance so that we can be sure the file system is instantiated before querying
+// for its mount name.
+inst.node.addDependency(fs);
+
+const mountPath = '/mnt/fsx';
+const dnsName = `${fs.fileSystemId}.fsx.${stack.region}.amazonaws.com`;
```

Comment:
This could have been a property on the `FsxFileSystem` class, would come in very useful. Don't forget to use `Aws.URL_SUFFIX` for the `amazonaws.com` part.

Reply:
I've added this property.